File: leetcode/0075-sort-colors/0075-sort-colors.ts
/**
 * @param nums
 * @returns
 */
function sortColors(nums: number[]): void {
    let red: number = 0;
    let white: number = 0;
    let blue: number = 0;

    for (const num of nums) {
        switch (num) {
            case 0:
                red++;
                break;
            case 1:
                white++;
                break;
            case 2:
                blue++;
                break;
        }
    }

    for (let i = 0; i < red; i++) {
        nums[i] = 0;
    }

    for (let i = red; i < red + white; i++) {
        nums[i] = 1;
    }

    for (let i = red + white; i < red + white + blue; i++) {
        nums[i] = 2;
    }
}

File: leetcode/0075-sort-colors/0075-sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        int red = 0;
        int white = 0;
        int blue = 0;

        for (int num : nums) {
            switch (num) {
                case 0:
                    red++;
                    break;
                case 1:
                    white++;
                    break;
                case 2:
                    blue++;
                    break;
            }
        }

        for (int i = 0; i < red; i++) {
            nums[i] = 0;
        }

        for (int i = red; i < red + white; i++) {
            nums[i] = 1;
        }

        for (int i = red + white; i < red + white + blue; i++) {
            nums[i] = 2;
        }
    }
}